process/transaction/shardProcess.go

```diff
@@ -169,35 +169,38 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) (vmco
 		if errors.Is(err, process.ErrInsufficientFunds) {
 			receiptErr := txProc.executingFailedTransaction(tx, acntSnd, err)
 			if receiptErr != nil {
+				// TODO: check if OK and receiptError is correct
 				return 0, receiptErr
 			}
 		}
 
 		if errors.Is(err, process.ErrUserNameDoesNotMatchInCrossShardTx) {
 			errProcessIfErr := txProc.processIfTxErrorCrossShard(tx, err.Error())
 			if errProcessIfErr != nil {
+				// TODO: check if OK and errProcessIfErr is correct
 				return 0, errProcessIfErr
 			}
+			// TODO: check if UserError and nil is correct
```

Comment:
yes. correct.

Reply:
in most of the cases just the error is verified, not the retunCode. Is it ok that the error is nil?

Reply:
Ok and nonNilError should work, but this could be different, because most of the times the error is checked.

Reply:
userError is returned back mostly for tests. if error is returned and the error is not ErrFailedTransaction -> than the transaction is of TYPE invalid and gets into the INVALID BLOCK. If other error - state is reverted to before the tx and the transaction is deleted from the pool - not added to the block.

Reply:
exactly, but with the case here, the returned error is nil, so the transaction is valid?

Reply:
ok!